File: TradeMaster/src/main/java/com/trademaster/stock/entity/Stock.java
package com.trademaster.stock.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("stock")
public class Stock {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String warehouseCode;

    private String warehouseName;

    private String goodsCode;

    private String goodsName;

    private String goodsSpec;

    private String skuCode;

    private String skuName;

    private String unit;

    private BigDecimal quantity;

    private BigDecimal lockedQuantity;

    private BigDecimal availableQuantity;

    private BigDecimal avgCostPrice;

    private BigDecimal totalAmount;

    private String batchNo;

    private LocalDate expiryDate;

    private Integer status;

    @TableLogic
    private Integer delFlag;

    private Long createBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    private Long updateBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: psi-stock-go/model/stock.go
package model

import "github.com/shopspring/decimal"

// StockEntity 库存实体（对应 psi_stock.stock）
type StockEntity struct {
	ID                int64           `db:"id"`
	TenantID          int64           `db:"tenant_id"`
	WarehouseCode     string          `db:"warehouse_code"`
	WarehouseName     string          `db:"warehouse_name"`
	GoodsCode         string          `db:"goods_code"`
	SkuCode           string          `db:"sku_code"`
	GoodsName         string          `db:"goods_name"`
	GoodsSpec         string          `db:"goods_spec"`
	Unit              string          `db:"unit"`
	Quantity          decimal.Decimal `db:"quantity"`
	LockedQuantity    decimal.Decimal `db:"locked_quantity"`
	AvailableQuantity decimal.Decimal `db:"available_quantity"`
	AvgCostPrice      decimal.Decimal `db:"avg_cost_price"`
	TotalAmount       decimal.Decimal `db:"total_amount"`
	Status            int             `db:"status"`
}

// StockOperateRequest 单商品操作请求
type StockOperateRequest struct {
	WarehouseCode string  `json:"warehouseCode" binding:"required"`
	GoodsCode     string  `json:"goodsCode"`
	SkuCode       string  `json:"skuCode" binding:"required"`
	Quantity      float64 `json:"quantity" binding:"required,gt=0"`
	CostPrice     float64 `json:"costPrice"`
	SourceNo      string  `json:"sourceNo"`
	SourceType    string  `json:"sourceType"`
	Remark        string  `json:"remark"`
}

// StockBatchOperateItem 批量操作项
type StockBatchOperateItem struct {
	WarehouseCode string  `json:"warehouseCode" binding:"required"`
	GoodsCode     string  `json:"goodsCode"`
	SkuCode       string  `json:"skuCode" binding:"required"`
	Quantity      float64 `json:"quantity" binding:"required,gt=0"`
}

// StockBatchOperateRequest 批量操作请求
type StockBatchOperateRequest struct {
	Items      []StockBatchOperateItem `json:"items" binding:"required,min=1"`
	SourceNo   string                  `json:"sourceNo"`
	SourceType string                  `json:"sourceType"`
	Remark     string                  `json:"remark"`
}
